crates/biome_js_analyze/src/semantic_analyzers/correctness/no_const_assign.rs

```diff
@@ -113,4 +116,58 @@ impl Rule for NoConstAssign {
             ),
         )
     }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let model = ctx.model();
+        let mut mutation = ctx.root().begin();
+
+        let declared_binding = model.binding(node)?;
+
+        if let Some(possible_declarator) = declared_binding.syntax().ancestors().find(|node| {
+            !AnyJsObjectBindingPatternMember::can_cast(node.kind())
+                && !JsObjectBindingPatternPropertyList::can_cast(node.kind())
+                && !AnyJsArrayBindingPatternElement::can_cast(node.kind())
+                && !JsArrayBindingPatternElementList::can_cast(node.kind())
+                && !JsIdentifierBinding::can_cast(node.kind())
+        }) {
+            if JsVariableDeclarator::can_cast(possible_declarator.kind()) {
+                let possible_declaration = possible_declarator.parent()?;
+                if let Some(js_variable_declaration) =
+                    JsVariableDeclaratorList::cast_ref(&possible_declaration)
+                        .and_then(|declaration| declaration.syntax().parent())
+                        .and_then(JsVariableDeclaration::cast)
+                {
+                    if js_variable_declaration.is_const() {
```

Comment:
Do we really need to check if it is `const`? It should be const? You can add a debug assert:

```rs
debug_assert!(js_variable_declaration.is_const(), "The declaration should be const.");
```